### DISCUSSION TITLE
How to use D1 database in a worker?

### DISCUSSION THREAD

**Discussion by qdubois** [skymethod/denoflare#58] [Q&A]
Did anyone accessed a D1 database within a worker in Deno ?

Thanks

Quentin

**Comment by johnspurlock-skymethod** [comment]
You should be able to use the [`denoflare d1`](https://denoflare.dev/cli/d1) cli (or the Cloudflare dashboard) to manage your dbs 

Then bind a d1 db to your worker via command line or config...

<img width="350" alt="image" src="https://github.com/skymethod/denoflare/assets/47259736/cdecebe0-93c9-41f9-9653-b4f3525287ee">

...then access it via the env at runtime, it will be typed as [D1Database](https://github.com/skymethod/denoflare/blob/b270f9e8c7f7b469d7a59329a18270f5c16b0b0a/common/cloudflare_workers_types.d.ts#L1228), reference the [D1 client API](https://developers.cloudflare.com/d1/platform/client-api/) from there
